internal/host/agent/agent_linux.go

```diff
@@ -303,62 +292,23 @@ func main() {
 
 	ctx := context.Background()
 
-	router := mux.NewRouter()
-
-	router.
-		Methods("PUT").
-		Path("/file/{name:.+}").
-		HandlerFunc(PutFileFn(ctx))
-	router.
-		Methods("GET").
-		Path("/file/{name:.+}").
-		HandlerFunc(GetFileFn(ctx))
-	router.
-		Methods("POST").
-		Path("/file/{name:.+}").
-		Headers("Content-Type", "application/yaml").
-		HandlerFunc(PostFileFn(ctx))
-	router.
-		Methods("DELETE").
-		Path("/file/{name:.+}").
-		HandlerFunc(DeleteFileFn(ctx))
-
-	router.
-		Methods("GET").
-		Path("/group/{name}").
-		HandlerFunc(GetGroupFn(ctx))
-
-	router.
-		Methods("GET").
-		Path("/user/{username}").
-		HandlerFunc(GetUserFn(ctx))
-
-	router.
-		Methods("GET").
-		Path("/ping").
-		HandlerFunc(GetPing)
-
-	router.
-		Methods("POST").
-		Path("/run").
-		Headers("Content-Type", "application/yaml").
-		HandlerFunc(PostRunFn(ctx))
-
-	router.
-		Methods("POST").
-		Path("/shutdown").
-		HandlerFunc(PostShutdownFn(ctx))
-
-	server := &http2.Server{}
-
-	serveConnOpts := &http2.ServeConnOpts{
-		Handler: h2c.NewHandler(router, server),
-	}
+	router := http.NewServeMux()
+
+	router.HandleFunc("PUT /file/{name}", PutFileFn(ctx))
+	router.HandleFunc("GET /file/{name}", GetFileFn(ctx))
+	router.HandleFunc("POST /file/{name}", PostFileFn(ctx))
+	router.HandleFunc("DELETE /file/{name}", DeleteFileFn(ctx))
+
+	router.HandleFunc("GET /group/{name}", GetGroupFn(ctx))
+	router.HandleFunc("GET /user/{username}", GetUserFn(ctx))
+
+	router.HandleFunc("GET /ping", GetPing)
+	router.HandleFunc("POST /run", PostRunFn(ctx))
+	router.HandleFunc("POST /shutdown", PostShutdownFn(ctx))
 
-	conn := aNet.Conn{
-		Reader: os.Stdin,
-		Writer: os.Stdout,
```

Comment:
I believe without something aking to this on the new server, it won't work, as it must have something binding the server to stdin/stdout.